package/MDAnalysis/coordinates/GSD.py

```diff
@@ -0,0 +1,129 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+"""GSD trajectory reader  --- :mod:`MDAnalysis.coordinates.GSD`
+============================================================
+
+Class to read the GSD trajectory, output of HOOMD-blue. The GSD format specifies
+both the topology and the trajectory of the particles in the simulation. The
+topology is read by the GSDTopology class.
+
+The GSD format was developed having in mind the possibility of changing number
+of particles, particle types, particle identities and changing topology.
+Currently this class has limited functionality, due to the fact that the number
+of particles and the topology are kept fixed in most MD simulations. The user
+will get an error only if at any time step the number of particles is detected
+to be different to the one that was set at the first time step. No check on
+changes in particle identity or topology is currently implemented.
+
+Classes
+-------
+
+.. autoclass:: GSDReader
+   :inherited-members:
+
+"""
+from __future__ import absolute_import, division
+
+import numpy as np
+from . import base
+try :
+    import gsd.hoomd
+except :
+    pass
```

Comment:
`gsd.hoomd` appears to be required in `GSDReader`. 

At the moment the [import policy](https://github.com/MDAnalysis/mdanalysis/wiki/Style-Guide#list-of-core-module-dependencies) is that there are no optional dependencies in the core library so this has to be a hard import that fails if this package is not installed.

If we want to do this differently then we need to discuss it.

Reply:
I think it's reasonable to have some formats not work out of the box if they have more difficult install routes (eg aren't on pypi).  We used to do this for netCDF before we started using scipy for that.